homeassistant/components/wolflink/__init__.py

```diff
@@ -113,6 +115,46 @@ async def async_update_data():
     hass.data[DOMAIN][entry.entry_id][COORDINATOR] = coordinator
     hass.data[DOMAIN][entry.entry_id][DEVICE_ID] = device_id
 
+    async def async_set_value(call):
+        """Handle the service call to write a value to the Wolf SmartSet system."""
```

Comment:
Can you elaborate more on what kind of values you write and for what purpose?

Reply:
you would find the specific values in the UI - it's a bit tricky to get them. The Addon "ism7mqtt" https://github.com/zivillian/ism7mqtt provides a list of (all) values:
<img width="1233" alt="grafik" src="https://github.com/user-attachments/assets/fd36b956-b087-4fb2-815e-f6ec0a65398f">

when you want to configure a value to set, you would look into the network-tab when you set the specific value over the original UI (and not this addon).

Reply:
No that's not what I mean, I mean, what are you expected to update with those values? Because they probably have a meaning and do something

Reply:
You can change the settings of your wolf device. Wolf has a complex mapping with value_id different per device. They go from installation setting till turning things on and off or putting the temperature higher. This would allow an advance user to automate their paramater from whatever wolf device to automate something.  Current integration only support reading and put everything as a sensor. This would enable for the first time home assistant to also change values back.

Reply:
Is it possible to get the context of such keys? As in, which keys represent a heating device etc

Reply:
Only when you initially parse the the json you get back from the API and correlated it to the structure. Wolfs own mapping lib is only availbile in C# It's a lot of reverse enginering as depending on your heating (gas/oil/heatpump) or ventilation or solar for hot water you get a different  json structure with different values back. I think that's why the orginal creator restricted it to heating systems only as it's the easiest. Also in the same API are all the Installer options. My personal goal is with this new feature in the lib which I introduced in 0.0.15 is to build eventualy switches etc, but still digging trhoug to figure out how I can figure it out constantly.

Reply:
With such JSON files, are you able to parse the context of all values?

Because the point I want to make is that ideally we don't add uncontextualized services. They require quite a lot of knowledge, and ideally the services used to modify a value have context. Meaning, I'd rather see a climate entity where you can set the temperature, than a generic service where you have to call register x with value y to achieve the same.

And yes, such PR would be something for the meantime, but I want to push back a bit to see if this can be figured out, since the project has a much bigger preference for entities than a generic service. The more mud we throw on the wall, the more we have to deprecate in the future.

Reply:
I agree. Technically part of the integration will need to be re-written to get the full support working as currently the integration doesn't discover individual devices and only takes the system yourself as the devicename on the internet and puts all the values into that for a few heating devices only. This solution would be interim solution to make it possible for advance users as currently it isn't.

See here a bit of snippet of the API JSON return. It's about 6x as big depending on example  setup (My example is consisting out of Solar and Gas heating). The valueId and some other parameters is what this service uses:

[{'IsExpertView': False, 'TabName': 'Übersicht', 'GuiId': 9500, 'BundleId': 1000, 'ParameterDescriptors': [{'ValueId': 18000500001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18000500001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'KF Kesselfühler', 'ControlType': 0, 'Value': '35.2', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18000800001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18000800001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'RLF Rücklauffühler', 'ControlType': 0, 'Value': '36.8', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18002500001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18002500001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'ZHP Heizkreispumpe', 'ControlType': 0, 'Value': '1', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18002600001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18002600001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': '3WUV 3-Wege-Umschaltventil', 'ControlType': 0, 'Value': '0', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18000600001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18000600001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'SF Speicherfühler', 'ControlType': 0, 'Value': '56.5', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18005400001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18005400001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'AGF Abgasfühler', 'ControlType': 0, 'Value': '34', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 18005100001, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 18005100001, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'DHK Druck Heizkreis', 'ControlType': 0, 'Value': '2.06', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 6003800000, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 6003800000, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'SKP Solarkreispumpe', 'ControlType': 0, 'Value': '0', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 6002500000, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 6002500000, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'SFS Solarfühler Speicher', 'ControlType': 0, 'Value': '45.6', 'ValueState': 0, 'HasDependentParameter': False}, {'ValueId': 6002600000, 'SortId': 0, 'SubBundleId': 0, 'ParameterId': 6002600000, 'IsReadOnly': False, 'NoDataPoint': False, 'IsExpertProtectable': False, 'Name': 'SFK Solarfühler Kollektor', 'ControlType': 0, 'Value': '17.5', 'ValueState': 0, 'HasDependentParameter': False}], 'ViewType': 7, 'SvgSchemaDeviceId': 0, 'GetValueLastAccess': '2024-10-24T16:17:21.1737712Z', 'TabViewGroups': [], 'SVGHeatingSchemaConfigDevices': [{'dt': 'HG', 'di': 1, 'deviceTemplateName': 'CGB_2', 'configIndex': 2, 'parameters': [{'valueId': 18000500001, 'unit': '°C', 'parameterName': 'KF Kesselfühler', 'parameterId': 18000500001}, {'valueId': 18000800001, 'unit': '°C', 'parameterName': 'RLF Rücklauffühler', 'parameterId': 18000800001}, {'valueId': 18002500001, 'parameterName': 'ZHP Heizkreispumpe', 'parameterId': 18002500001}, {'valueId': 18002600001, 'parameterName': '3WUV 3-Wege-Umschaltventil', 'parameterId': 18002600001}, {'valueId': 18000600001, 'unit': '°C', 'parameterName': 'SF Speicherfühler', 'parameterId': 18000600001}, {'valueId': 18005400001, 'unit': '°C', 'parameterName': 'AGF